src/components/admin/Settings/PrefixAlerts/Dialog/SaveButton.tsx

```diff
@@ -0,0 +1,109 @@
+import { LoadingButton } from '@mui/lab';
+import {
+    createNotificationSubscription,
+    deleteNotificationSubscription,
+} from 'api/alerts';
+import { EmailDictionary } from 'components/admin/Settings/PrefixAlerts/types';
+import { useZustandStore } from 'context/Zustand/provider';
+import { Dispatch, SetStateAction, useState } from 'react';
+import { FormattedMessage } from 'react-intl';
+import {
+    SelectableTableStore,
+    selectableTableStoreSelectors,
+} from 'stores/Tables/Store';
+import { SelectTableStoreNames } from 'stores/names';
+import { hasLength } from 'utils/misc-utils';
+
+interface Props {
+    disabled: boolean;
+    existingEmails: EmailDictionary;
+    prefix: string;
+    setOpen: Dispatch<SetStateAction<boolean>>;
+    updatedEmails: EmailDictionary;
+}
+
+function SaveButton({
+    disabled,
+    existingEmails,
+    prefix,
+    setOpen,
+    updatedEmails,
+}: Props) {
+    const hydrate = useZustandStore<
+        SelectableTableStore,
+        SelectableTableStore['hydrate']
+    >(
+        SelectTableStoreNames.PREFIX_ALERTS,
+        selectableTableStoreSelectors.query.hydrate
+    );
+
+    const [loading, setLoading] = useState(false);
+
+    const onClick = async (event: React.MouseEvent<HTMLElement>) => {
+        event.preventDefault();
+        setLoading(true);
+
+        const subscriptionsToCreate: [string, string][] = [];
+
+        Object.entries(updatedEmails).forEach(([key, value]) => {
+            const processedValues = Object.hasOwn(existingEmails, key)
+                ? value.filter((email) => !existingEmails[key].includes(email))
+                : value;
+
+            processedValues
+                .map((email): [string, string] => [key, email])
+                .forEach((subscriptionMetadata) => {
+                    subscriptionsToCreate.push(subscriptionMetadata);
+                });
+        });
+
+        const createdSubscriptions = subscriptionsToCreate.map(([key, email]) =>
+            createNotificationSubscription(key, email)
+        );
+
+        const subscriptionsToCancel: [string, string][] = [];
+
+        Object.entries(updatedEmails).forEach(([key, value]) => {
+            if (Object.hasOwn(existingEmails, key)) {
+                existingEmails[key]
+                    .filter((email) => !value.includes(email))
+                    .map((email): [string, string] => [key, email])
+                    .forEach((subscriptionMetadata) => {
+                        subscriptionsToCancel.push(subscriptionMetadata);
+                    });
+            }
+        });
```

Comment:
`todo`
Looking at this... I do not think we need to have two loops here. Can we try to combine these two loops to reduce how often we have to loop. Do not need to get handled with this PR so either it can get fixed or mark it with a `todo`

Reply:
Ha, I cannot believe I missed this on my first pass at refactoring. The logic used to be unique here. There is no need for the to loops over the `updatedEmails` object. Will update.